WireUI/Sources/WireMainNavigationUI/Coordinator/MainCoordinator.swift

```diff
@@ -197,24 +193,26 @@ public final class MainCoordinator<Dependencies>: NSObject, MainCoordinatorProto
     public func showConversation(
         conversation: ConversationModel,
         message: ConversationMessageModel?
-    ) async {
-        if mainSplitViewState == .expanded, splitViewController.splitBehavior == .overlay {
-            splitViewController.hideSidebar()
-        }
-
-        await dismissPresentedViewController()
+    ) {
+        Task { @MainActor in
```

Comment:
In my opinion marking it async is like a promise to the caller that it has completely finished when await returns. However, since `UINavigationController.pushViewController(_:animated:)` and `setViewControllers(_:animated:)` don't offer a completion block, this method can't make this promise.

What's your opinion?
1. keep it like I changed it
2. make it async anyway
3. use the `transitionCoordinator` in order to await the navigation controller's animations? (see UINavigationController+Additions.swift)
@netbe

Reply:
1. for now keep it as it is